apps/veyfi/utils/actions/votingEscrow.ts

```diff
@@ -0,0 +1,95 @@
+import {ethers} from 'ethers';
+import {approveERC20} from '@common/utils/actions/approveToken';
+
+import VEYFI_ABI from '../abi/veYFI.abi';
+
+import type {BigNumber} from 'ethers';
+import type {TAddress} from '@yearn-finance/web-lib/utils/address';
+import type {TSeconds} from '@yearn-finance/web-lib/utils/time';
+
+const handleTx = async (txPromise: Promise<ethers.providers.TransactionResponse>): Promise<boolean> => {
+	try {
+		const tx = await txPromise;
+		const receipt = await tx.wait();
+		if (receipt.status === 0) {
+			console.error('Fail to perform transaction');
+			return false;
+		}
```

Comment:
As we have a catch block that does exactly this, I'd just throw here
```suggestion
		if (receipt.status === 0) {
			throw new Error('Fail to perform transaction');
		}
```

Reply:
That would be better yes. Although in this particular function there's a catch block that logs the error and returns `false`, so was suggesting we can just throw inside the `try` block so that it's caught in the next block